src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs

```diff
@@ -121,20 +114,30 @@ private void AttachHandlers(IRepository repository)
         {
             if (repository == null)
                 return;
-            repository.OnHeadChanged += Refresh;
-            repository.OnStatusUpdated += UpdateStatusOnMainThread;
-            repository.OnActiveBranchChanged += s => Refresh();
-            repository.OnActiveRemoteChanged += s => Refresh();
+            repository.OnLocalBranchChanged += Refresh;
+            repository.OnStatusChanged += UpdateStatusOnMainThread;
+            repository.OnCurrentBranchChanged += Repository_OnCurrentBranchChanged();
+            repository.OnCurrentRemoteChanged += Repository_OnCurrentRemoteChanged();
         }
 
         private void DetachHandlers(IRepository repository)
         {
             if (repository == null)
                 return;
-            repository.OnHeadChanged -= Refresh;
-            repository.OnStatusUpdated -= UpdateStatusOnMainThread;
-            repository.OnActiveBranchChanged -= s => Refresh();
-            repository.OnActiveRemoteChanged -= s => Refresh();
+            repository.OnLocalBranchChanged -= Refresh;
+            repository.OnStatusChanged -= UpdateStatusOnMainThread;
+            repository.OnCurrentBranchChanged -= Repository_OnCurrentBranchChanged();
+            repository.OnCurrentRemoteChanged -= Repository_OnCurrentRemoteChanged();
+        }
+
+        private Action<string> Repository_OnCurrentRemoteChanged()
+        {
+            return s => Refresh();
+        }
+
+        private Action<string> Repository_OnCurrentBranchChanged()
+        {
+            return s => Refresh();
```

Comment:
You can't call Refresh directly here, this is triggered from a different thread, you need to shuffle it off to the UI thread.